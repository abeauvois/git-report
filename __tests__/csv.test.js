const { toArray } = require("rxjs/operators");
const { fromCSVFile } = require("../src/fromCSVFile");

let received, expected, result;

describe("csv", () => {
  beforeEach(() => {
    received = null;
    expected = null;
  });
  test("R1: fromCSVFile", (done) => {
    result = null;
    fromCSVFile("./__tests__/test_file.csv")
      .pipe(toArray())
      .subscribe(
        (received) => {
          expect(received).toMatchSnapshot();
          done();
        },
        (e) => e
      );
  });
});
