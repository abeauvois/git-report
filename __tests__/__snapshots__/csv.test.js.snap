// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv R1: fromCSVFile 1`] = `
Array [
  Object {
    "contributor": " contributor",
    "date": " date",
    "message": " message",
    "sha": "sha",
  },
  Object {
    "contributor": " abeauvois",
    "date": " Fri Nov 13 10:40:39 2020",
    "message": " \\"m1 nov\\"",
    "sha": "1",
  },
  Object {
    "contributor": " abeauvois",
    "date": " Thu Nov 12 16:06:45 2020",
    "message": " \\"m2 nov\\"",
    "sha": "2",
  },
  Object {
    "contributor": " abeauvois",
    "date": " Thu Nov 12 16:03:21 2020",
    "message": " \\"m3 nov\\"",
    "sha": "3",
  },
  Object {
    "contributor": " abeauvois",
    "date": " Thu Oct 12 16:03:21 2020",
    "message": " \\"m2 oct\\"",
    "sha": "3",
  },
  Object {
    "contributor": " abeauvois",
    "date": " Thu Sep 12 16:03:21 2020",
    "message": " \\"m1 sep\\"",
    "sha": "3",
  },
]
`;
